assets/js/hooks/useLatestIntersection.test.js

```diff
@@ -0,0 +1,258 @@
+		const { result, rerender } = renderHook( () =>
+			useLatestIntersection( targetRef, {
+				root: container,
+				threshold: 0.8,
+			} )
+		);
+
+		const mockIntersectionObserverEntry = {
+			boundingClientRect: targetRef.current.getBoundingClientRect(),
+			intersectionRatio: 0.81,
+			intersectionRect: container.getBoundingClientRect(),
+			isIntersecting: true,
+			rootBounds: container.getBoundingClientRect(),
+			target: targetRef.current,
+			time: 300,
+		};
+		actHook( () => {
+			intersectionObserver.simulate( mockIntersectionObserverEntry );
+		} );
+
+		expect( result.current ).toEqual( mockIntersectionObserverEntry );
+
+		targetRef.current = document.createElement( 'div' );
+		rerender();
+
+		expect( result.current ).toEqual( null );
+	} );
+
+	it( 'should return null if IntersectionObserver is not supported', () => {
+		targetRef = createRef();
+		targetRef.current = document.createElement( 'div' );
+		delete global.IntersectionObserver;
+
+		expect( () =>
+			renderHook( () => useLatestIntersection( targetRef, {} ) )
+		).not.toThrow();
+	} );
+
+	it( 'should disconnect an old IntersectionObserver instance when the ref changes', () => {
+		targetRef = createRef();
+		targetRef.current = document.createElement( 'div' );
+
+		const { rerender } = renderHook( () =>
+			useLatestIntersection( targetRef, {} )
+		);
+
+		targetRef.current = document.createElement( 'div' );
+		rerender();
+
+		targetRef.current = null;
+		rerender();
+
+		expect( IntersectionObserver ).toHaveBeenCalledTimes( 2 );
+		expect(
+			IntersectionObserver.prototype.disconnect
+		).toHaveBeenCalledTimes( 2 );
+	} );
+
+	it( 'should return the last IntersectionObserverEntry when the IntersectionObserver registers an intersection', () => {
+		act( () => {
+			targetRef = createRef();
+			render( <div ref={ targetRef } />, container );
+		} );
+
+		const { result } = renderHook( () =>
+			useLatestIntersection( targetRef, {
+				root: container,
+				threshold: 0.3,
+			} )
+		);
+
+		const mockIntersectionObserverEntry1 = {
+			boundingClientRect: targetRef.current.getBoundingClientRect(),
+			intersectionRatio: 0.31,
+			intersectionRect: container.getBoundingClientRect(),
+			isIntersecting: true,
+			rootBounds: container.getBoundingClientRect(),
+			target: targetRef.current,
+			time: 300,
+		};
+
+		actHook( () => {
+			intersectionObserver.simulate( mockIntersectionObserverEntry1 );
+		} );
+
+		expect( result.current ).toEqual( mockIntersectionObserverEntry1 );
+
+		const mockIntersectionObserverEntry2 = {
+			boundingClientRect: targetRef.current.getBoundingClientRect(),
+			intersectionRatio: 0.61,
+			intersectionRect: container.getBoundingClientRect(),
+			isIntersecting: true,
+			rootBounds: container.getBoundingClientRect(),
+			target: targetRef.current,
+			time: 400,
+		};
+
+		actHook( () => {
+			intersectionObserver.simulate( mockIntersectionObserverEntry2 );
+		} );
+
+		expect( result.current ).toEqual( mockIntersectionObserverEntry2 );
```

Comment:
This test should be improved by dispatching both entries together, to ensure that the [`entries`](https://github.com/google/site-kit-wp/blob/155704296821ea408be5c26be2cf74b4ebc00e66/assets/js/hooks/useLatestIntersection.js#L39-L41) array has got more than one item in it, verifying that it's the last item being returned.

```suggestion
		const mockIntersectionObserverEntry2 = {
			boundingClientRect: targetRef.current.getBoundingClientRect(),
			intersectionRatio: 0.61,
			intersectionRect: container.getBoundingClientRect(),
			isIntersecting: true,
			rootBounds: container.getBoundingClientRect(),
			target: targetRef.current,
			time: 400,
		};

		actHook( () => {
			intersectionObserver.simulate( [
				mockIntersectionObserverEntry1,
				mockIntersectionObserverEntry2,
			] );
		} );

		expect( result.current ).toEqual( mockIntersectionObserverEntry2 );
```